### ISSUE TITLE
Allow the loading order of IEA data to be prioritised

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#1149]
so that we can load in a preferred order:

GO_REF:0000003	IEA	EC
GO_REF:0000116	IEA	RHEA
GO_REF:0000041	IEA	UniPathway

GO_REF:0000002	IEA	InterPro
GO_REF:0000104	IEA	UniRule
GO_REF:0000043	IEA	UniProtKB-KW
GO_REF:0000044	IEA	UniProtKB-SubCell

GO_REF:0000117	IEA	ARBA
TreeGrafter which isn't currently loaded.

**Comment by kimrutherford** [comment]
There's a prototype of this running on [my desktop](https://desktop.kmr.nz/gene/SPBC16H5.06).  I hope I got the priority right with EC being the highest priority and ARBA being the lowest.  Currently the GO_REFS not in your list have higher priority than those in your list.  Let me know if I should change that.

Here are GO_REFs that are use in GO annotations but are not in your list:
 - GO_REF:00000051
 - GO_REF:0000024
 - GO_REF:0000033
 - GO_REF:0000036
 - GO_REF:0000050
 - GO_REF:0000051
 - GO_REF:0000111
 - GO_REF:0000114

There are about 360 annotation differences after making these filtering changes.

----------------
Before - https://www.pombase.org/gene/SPBC16H5.06
(Might look different after the nightly load)

![image](https://github.com/pombase/pombase-chado/assets/90474/f2d15068-9d83-4ecd-adf7-c8fbb102ebd4)

----------------------------
After the code change - https://desktop.kmr.nz/gene/SPBC16H5.06

![image](https://github.com/pombase/pombase-chado/assets/90474/5cd5cfcc-a419-4418-909b-686bbdc19a8e)

**Comment by ValWood** [comment]
This looks good,


    GO_REF:00000051 is a typo, I will fix

These should not be used for IEA, ar they? If so I need to fix
    GO_REF:0000024
    GO_REF:0000033
    GO_REF:0000036
    GO_REF:0000050
    GO_REF:0000051
    GO_REF:0000111
    GO_REF:0000114


ALso can I change the order slightly
Rhea then EC.
We should align with Rhea first because GO is aligned with Rhea.

Also for tonight (or tomorrow)  could you include TReeGrafter  as the lowest priority.
We might take it out again, but I need the numbers for the paper and I can have another look

**Comment by ValWood** [comment]
yes go ahead, sorry for delay, in transit back south

**Comment by kimrutherford** [comment]
> Also for tonight (or tomorrow) could you include TReeGrafter as the lowest priority.

I've added the TreeGrafter data to tonight's load and added [GO_REF:0000118](https://github.com/geneontology/go-site/blob/master/metadata/gorefs/goref-0000118.md) to the IEA GO filtering.

The filtering changes are committed for tonight.

**Comment by kimrutherford** [comment]
The load has finished.  There are about 1300 more GO annotations than yesterday and about 1500 TreeGrafter annotations: https://pombase.org/reference/GO_REF:0000118

My desktop is running the Chado from yesterday if you'd like to compare anything.

**Comment by ValWood** [comment]
Great.
Can you remon me the order we are currently using to filter redundancy?

**Comment by kimrutherford** [comment]
```
    inferred from physical interaction   # special case: only when missing "with"
    inferred from biological aspect of ancestor
    inferred from biological aspect of descendant
    inferred by curator
    inferred from sequence orthology
    inferred from sequence or structural similarity
    inferred from sequence model
    traceable author statement
    non-traceable author statement
    inferred from expression pattern
    inferred from electronic annotation
    inferred from reviewed computational analysis
```

**Comment by ValWood** [comment]
a) do we have any with "inferred from reviewed computational analysis", I think this is legacy, in which case we can remove

b) could you do one run with the precedence 

inferred from physical interaction   # special case: only when missing "with"
 
    inferred from biological aspect of descendant
    inferred by curator
    inferred from sequence orthology
    inferred from sequence or structural similarity
    inferred from sequence model
    traceable author statement
    non-traceable author statement
    inferred from expression pattern
*****   inferred from biological aspect of ancestor
    inferred from electronic annotation

(It won't change actual annotations but it will allow me to see how many. manual are covered by IBA)

**Comment by kimrutherford** [comment]
>  do we have any with "inferred from reviewed computational analysis", I think this is legacy, in which case we can remove

Yep, I can't see any in Chado.  I'll remove it.

**Comment by kimrutherford** [comment]
> could you do one run with the precedence 

So temporarily move IBA to just above IEA in precedence?

I'll run it on my desktop while the main load is running tonight.  Then it will be easy to compare.

**Comment by ValWood** [comment]
Yes that's correct. I forgot that we gave it higher precedence, and I want to see how many of the annotations are covered by any manual code.

**Comment by kimrutherford** [comment]
> temporarily move IBA to just above IEA in precedence?

Hi Val.

Here are the before and after GAF files with the extra product and term name columns: 

 - https://curation.pombase.org/kmr44/pombase_style_gaf-2024-03-13-standard-ev-order.tsv
 - https://curation.pombase.org/kmr44/pombase_style_gaf-2024-03-13-new-ev-order.tsv

**Comment by kimrutherford** [comment]
> perfect, I got the number, we can switch back to standard.

I didn't change the code for the nightly load, I run it on my desktop with the order switched.

So pombase.org has GO annotations with the normal filter order.

**Comment by kimrutherford** [comment]
Is there anything else to do here?

**Comment by ValWood** [comment]
NO, I might need a run with a different order at some point to extract an intersection. I think I know how to get this but will discuss with you at some point! no hurry, I can do a back of an envelope figure to demonstrate my point.

**Comment by ValWood** [comment]
Tha wasn't meant to be a shouty NO ;)
